tools/c7n_azure/c7n_azure/session.py

```diff
@@ -353,3 +303,160 @@ def get_functions_auth_string(self, target_subscription_id):
                 "supported auth mechanism for deploying functions.")
 
         return json.dumps(function_auth_params, indent=2)
+
+
+@six.add_metaclass(abc.ABCMeta)
+class TokenProvider:
+    AuthenticationResult = namedtuple('AuthenticationResult', 'token, subscription_id, tenant_id')
```

Comment:
Looking at all usages and I think it makes sense to keep name `credentials` instead of `token`. What do you think?

Reply:
Oh I agree for sure, but this line is 99 characters and I am so tired of unreadable code due to excessive line breaks that I choose sub-optimal names to avoid it when possible :(